src/sql/workbench/browser/designer/designer.ts

```diff
@@ -941,6 +983,56 @@ export class Designer extends Disposable implements IThemable {
 		return component;
 	}
 
+	private addTableDesignerTaskbar(container: HTMLElement, tableProperties: DesignerTableProperties,
+		path: DesignerPropertyPath, view: DesignerUIArea): void {
+
+		let needButtons: boolean = tableProperties.canAddRows || tableProperties.canMoveRows;
+		if (needButtons) {
+			const taskbarContainer = container.appendChild(DOM.$('.full-row')).appendChild(DOM.$('.add-row-button-container'));
+
+			this._actionBar = new Taskbar(taskbarContainer);
+			let actions: IAction[] = [];
+
+			if (tableProperties.canAddRows) {
+				let addColAction = this._instantiationService.createInstance(AddColumnAction, this, tableProperties);
+				actions.push(addColAction);
+			}
+			if (tableProperties.canMoveRows) {
+				let moveUpAction = this._instantiationService.createInstance(MoveRowUpAction, this);
+				let moveDownAction = this._instantiationService.createInstance(MoveRowDownAction, this);
+				actions.push(moveUpAction);
+				actions.push(moveDownAction);
+			}
+			let taskbarContent: ITaskbarContent[] = actions.map((a) => { return { action: a }; });
+			this._actionBar.setContent(taskbarContent);
+			let edit: DesignerEdit = { type: undefined, path: path, source: view };
+			this._actionBar.context = edit;
```

Comment:
is this really needed?

Reply:
yes

Reply:
why? I can understand the context being the table information. you can always construct a DesignerEdit object when you need it in the action handler.

Reply:
but we need the path and source for it, which we dont have

Reply:
you can define an interface DesignerActionBarContext